Assets/BossRoom/Scripts/Client/UI/Lobby/JoinLobbyUI.cs

```diff
@@ -0,0 +1,160 @@
+using System.Collections.Generic;
+using BossRoom.Scripts.Shared.Infrastructure;
+using BossRoom.Scripts.Shared.Net.UnityServices.Lobbies;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace BossRoom.Scripts.Client.UI
+{
+    /// <summary>
+    /// Handles the list of LobbyButtons and ensures it stays synchronized with the lobby list from the service.
+    /// </summary>
+    public class JoinLobbyUI : ObserverBehaviour<LobbyServiceData>
+    {
+        private UIFactory m_UIFactory;
+        private LobbyUIManager m_LobbyUIManager;
+        private LobbyServiceData m_LobbyServiceData;
+
+        [SerializeField] private LobbyPanelUI m_LobbyPanelPrototype;
+
+        //
+        [SerializeField]
+        InputField m_JoinCodeField;
+
+        /// <summary>
+        /// Key: Lobby ID, Value Lobby UI
+        /// </summary>
+        Dictionary<string, LobbyPanelUI> m_LobbyButtons = new Dictionary<string, LobbyPanelUI>();
+        Dictionary<string, LocalLobby> m_LocalLobby = new Dictionary<string, LocalLobby>();
+
+        /// <summary>Contains some amount of information used to join an existing lobby.</summary>
+        LocalLobby.LobbyData m_LocalLobbySelected;
+
+
+        [Inject]
+        private void InjectDependencies(UIFactory uiFactory, LobbyUIManager lobbyUIManager, LobbyServiceData lobbyServiceData)
+        {
+            m_UIFactory = uiFactory;
+            m_LobbyUIManager = lobbyUIManager;
+            m_LobbyServiceData = lobbyServiceData;
+        }
+
+        private void Awake()
+        {
+            m_LobbyPanelPrototype.gameObject.SetActive(false);
+        }
+
+        public void LobbyPanelSelected(LocalLobby lobby)
+        {
+            m_LocalLobbySelected = lobby.Data;
+        }
+
+        public void OnLobbyCodeInputFieldChanged(string newCode)
+        {
+            if (!string.IsNullOrEmpty(newCode))
+                m_LocalLobbySelected = new LocalLobby.LobbyData(newCode.ToUpper());
```

Comment:
Does this event trigger on every single character entered into the field?  Do we actually need to create the lobby instance when the field changes or can we just do it when the join button is pressed.